src/main/java/space/obminyashka/items_exchange/rest/LocationController.java

```diff
@@ -56,6 +57,21 @@ public List<LocationNameView> getAllAreas() {
         return locationService.findAllAreas();
     }
 
+    @GetMapping(value = ApiKey.LOCATION_DISTRICT, produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get all of existed districts by area id.")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
+            @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+    @ResponseStatus(HttpStatus.OK)
+    public List<LocationNameView> getAllDistricts(@RequestParam UUID areaId) throws EntityIdNotFoundException {
+        List<LocationNameView> locationNameViewList = locationService.findAllDistrictsByAreaId(areaId);
+        if (locationNameViewList.isEmpty()) {
+            throw new EntityIdNotFoundException(getParametrizedMessageSource(INVALID_LOCATION_ID, areaId));
+        }
```

Comment:
Looks a bit odd.
It'd be much simpler to check if an area with such ID exists and throw 404 exception if not